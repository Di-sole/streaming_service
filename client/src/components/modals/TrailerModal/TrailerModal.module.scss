.modal {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}
.container {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}
@media (max-width: 750px) {
    .container {
        height: 70%;
    }
}
