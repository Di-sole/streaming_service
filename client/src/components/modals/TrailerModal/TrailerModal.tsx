import ModalUI from '@/components/UI/Modal/ModalUI';
import styles from './TrailerModal.module.scss';
import dynamic from 'next/dynamic';

interface TrailerModalProps {
    trailer: string;
}

const TrailerModal = ({ trailer }: TrailerModalProps) => {
    const ReactPlayer = dynamic(() => import('react-player/lazy'), { ssr: false });
    return (
        <ModalUI className={styles.modal}>
            <div className={styles.container}>
                {trailer ? (
                    <ReactPlayer className={styles.player} width="80%" height="80%" url={trailer} controls={true} />
                ) : (
                    <h2>Трейлер еще не появился</h2>
                )}
            </div>
        </ModalUI>
    );
};

export default TrailerModal;
