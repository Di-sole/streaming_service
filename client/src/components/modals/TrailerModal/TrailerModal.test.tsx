import { render, screen } from '@testing-library/react';
import TrailerModal from './TrailerModal';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            asPath: '/',
        };
    },
}));

describe('TRAILERMODAL SNAPSHOTS TESTS', () => {
    test('TrailerModal should not be changed', () => {
        render(<TrailerModal trailer="" />);

        const trailer = screen.getByTestId('trailerModal');
        expect(trailer).toMatchSnapshot();
    });
});
