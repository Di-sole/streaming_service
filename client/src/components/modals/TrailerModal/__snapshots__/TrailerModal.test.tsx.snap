// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TRAILERMODAL SNAPSHOTS TESTS TrailerModal should not be changed 1`] = `
<div
  class="container"
  data-testid="trailerModal"
>
  <h2>
    Trailer not yet added
  </h2>
</div>
`;
